pkgs/build-support/rust/build-rust-package/default.nix

```
@@ -112,16 +110,16 @@ lib.extendMkDerivation {
           throw "cargoHash, cargoVendorDir, cargoDeps, or cargoLock must be set"
         else
           fetchCargoVendor (
-            {
-              inherit
-                src
-                srcs
-                sourceRoot
-                cargoRoot
-                preUnpack
-                unpackPhase
-                postUnpack
-                ;
+            getOptionalAttrs [
+              "src"
+              "srcs"
+              "sourceRoot"
+              "cargoRoot"
+              "preUnpack"
+              "unpackPhase"
+              "postUnpack"
+            ] finalAttrs
+            // {
               name = cargoDepsName;
               patches = cargoPatches;
               hash = args.cargoHash;
```

```suggestion
              hash = finalAttrs.cargoHash;
```

Also change `else if args.cargoHash or null == null then` to `else if finalAttrs.cargoHash or null == null then`